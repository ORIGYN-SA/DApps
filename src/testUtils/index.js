import React from 'react';
import TestEnvironment from './testEnv'

export * from '@testing-library/react';

export const makeFormikMethods = (mockFn) => ({
  resetForm: mockFn,
  setErrors: mockFn,
  setFieldError: mockFn,
  setFieldTouched: mockFn,
  setFieldValue: mockFn,
  setFormikState: mockFn,
  setStatus: mockFn,
  setSubmitting: mockFn,
  setTouched: mockFn,
  setValues: mockFn,
  submitForm: mockFn,
  validateField: mockFn,
  validateForm: mockFn,
});

module.exports = TestEnvironment