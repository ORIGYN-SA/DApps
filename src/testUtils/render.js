import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { createMemoryHistory } from 'history';

function customRender(children) {
    const history = createMemoryHistory({ initialEntries: ['/'] });
    return render(<Router history={history}>{children}</Router>);
}

export { customRender as render };