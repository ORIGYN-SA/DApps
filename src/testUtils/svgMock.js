import React from 'react';

const SvgMock = () => <svg></svg>;

export default SvgMock;
