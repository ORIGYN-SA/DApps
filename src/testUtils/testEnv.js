import Environment from 'jest-environment-jsdom-global'
class TestEnvironment extends Environment {
  constructor(config, context) {
    super(config, context);
    if (typeof this.global.TextEncoder === 'undefined') {
      const { TextEncoder } = require('util');
      this.global.TextEncoder = TextEncoder;
    }
    if (typeof this.global.TextDecoder === 'undefined') {
      const { TextDecoder } = require('util');
      this.global.TextDecoder = TextDecoder;
    }
    if (typeof this.global.fetch === 'undefined') {
      const fetch = require('isomorphic-fetch');
      this.global.fetch = fetch;
    }
  }
};

module.exports = TestEnvironment