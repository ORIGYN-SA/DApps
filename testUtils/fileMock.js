import module from 'module';
module.exports = 'test-file-stub';
