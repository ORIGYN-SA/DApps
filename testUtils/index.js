import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'

import { createMemoryHistory } from 'history'

function customRender(children) {
  function Wrapper({ children }) {
    const history = createMemoryHistory({ initialEntries: ['/'] })
    return <Router history={history}>{children}</Router>
  }
  return render(<Wrapper>{children}</Wrapper>)
}
export * from '@testing-library/react'

export { customRender as render }

export const makeFormikMethods = (mockFn) => ({
  resetForm: mockFn,
  setErrors: mockFn,
  setFieldError: mockFn,
  setFieldTouched: mockFn,
  setFieldValue: mockFn,
  setFormikState: mockFn,
  setStatus: mockFn,
  setSubmitting: mockFn,
  setTouched: mockFn,
  setValues: mockFn,
  submitForm: mockFn,
  validateField: mockFn,
  validateForm: mockFn,
})
