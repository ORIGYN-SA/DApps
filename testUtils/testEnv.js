const Environment = require('jest-environment-jsdom-global');

module.exports = class CustomTestEnvironment extends Environment {
  constructor({ globalConfig, projectConfig }, context) {
    super({ globalConfig, projectConfig }, context);
    if (typeof this.global.TextEncoder === 'undefined') {
      const { TextEncoder } = require('util');
      this.global.TextEncoder = TextEncoder;
    }
    if (typeof this.global.fetch === 'undefined') {
      const fetch = require('node-fetch');
      this.global.fetch = fetch;
    }
  }
};
