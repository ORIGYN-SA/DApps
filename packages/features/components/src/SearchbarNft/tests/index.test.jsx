import React, { useContext, useEffect, useState } from 'react';
import { render, fireEvent, waitFor, screen } from '../../../../../../testUtils';
import '@testing-library/jest-dom';
import { AuthContext } from '@dapp/features-authentication';
import { SearchbarNft } from '../index';

describe('Select-Autocomplete', () => {
  // Render
  render(<SearchbarNft isLoading={true}/>);
  // Test1
  const Loader = screen.getByRole('progressbar');
  test('If is loading should display only preloader', () => {
    expect(Loader).toBeInTheDocument();
  }
  );
  // Test2
  test('If is not loading should display the SEARCHBAR', () => {
    render(<SearchbarNft isLoading={false}/>);
    const Searchbar = screen.getByLabelText('Other tokens IDS');
    expect(Searchbar).toBeInTheDocument();
  }
  );
  // Test3
  // test searchbar is rendered and match snapshot
  test('Searchbar is rendered and match snapshot', () => {
    const { asFragment } = render(<SearchbarNft isLoading={false}/>);
    expect(asFragment()).toMatchSnapshot();
  }
  );
  // Test4
  //pass all the properties to the component
  test('Searchbar is rendered with all the properties', () => {
    const { asFragment } = render(<SearchbarNft isLoading={false} searchBarTokenId={'#myId'} />);
    expect(asFragment()).toMatchSnapshot();
  }
  );
  
  // Test5
  // if autocomplete is open and user click on the item, the item should be selected
  test('If autocomplete is open and user click on the item, the item should be selected', async () => {
    const { asFragment } = render(<SearchbarNft isLoading={false} searchBarTokenId={'#myId'} />);
    const Searchbar = screen.getByLabelText('Other tokens IDS');
    fireEvent.focus(Searchbar);
    const Item = screen.getByText('#myId');
    fireEvent.click(Item);
    expect(Searchbar.value).toBe('#myId');
  });

  // Test6
  // if the token id is not selected, the searchbar value should be 'Not selected'
  test('If the token id is not selected, the searchbar value should be \'Not selected\'', async () => {
    const { asFragment } = render(<SearchbarNft isLoading={false} searchBarTokenId={'Not selected'} />);
    const Searchbar = screen.getByLabelText('Other tokens IDS');
    expect(Searchbar.value).toBe('Not selected');
  }  
  );

  // Test7 
  // when user click on the searchbar, the autocomplete should be open
  test('When user click on the searchbar, the autocomplete should be open', async () => {
    const { asFragment } = render(<SearchbarNft isLoading={false} searchBarTokenId={'Not selected'} />);
    const Searchbar = screen.getByLabelText('Other tokens IDS');
    fireEvent.focus(Searchbar);
    expect(Searchbar.value).toBe('Not selected');
  });
});
