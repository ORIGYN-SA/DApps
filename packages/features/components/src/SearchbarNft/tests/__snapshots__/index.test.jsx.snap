// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select-Autocomplete Searchbar is rendered and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiBox-root css-1gpybnd-MuiPaper-root"
  >
    <div
      class="MuiFormControl-root css-78me8v-MuiFormControl-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-17r5iss-MuiTypography-root"
      >
        Current Token ID: 
        <b />
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1mi2yzt-MuiTypography-root"
      >
        Search for other NFT'S 
        <em>
          (+...)
        </em>
      </p>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-60y2nb-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="combo-box-demo"
            id="combo-box-demo-label"
          >
            Other tokens IDS
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="combo-box-demo"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Other tokens IDS
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select-Autocomplete Searchbar is rendered with all the properties 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiBox-root css-1gpybnd-MuiPaper-root"
  >
    <div
      class="MuiFormControl-root css-78me8v-MuiFormControl-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-17r5iss-MuiTypography-root"
      >
        Current Token ID: 
        <b>
          #myId
        </b>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1mi2yzt-MuiTypography-root"
      >
        Search for other NFT'S 
        <em>
          (+...)
        </em>
      </p>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-60y2nb-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="combo-box-demo"
            id="combo-box-demo-label"
          >
            Other tokens IDS
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="combo-box-demo"
              role="combobox"
              spellcheck="false"
              type="text"
              value="#myId"
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1in441m"
              >
                <span>
                  Other tokens IDS
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select-Autocomplete Test the collectionName is passed to the component 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiBox-root css-1gpybnd-MuiPaper-root"
  >
    <div
      class="MuiFormControl-root css-78me8v-MuiFormControl-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-17r5iss-MuiTypography-root"
      >
        Current Token ID: 
        <b />
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1mi2yzt-MuiTypography-root"
      >
        Search for other NFT'S 
        <em>
          (+...)
        </em>
      </p>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-60y2nb-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="combo-box-demo"
            id="combo-box-demo-label"
          >
            Other tokens IDS
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="combo-box-demo"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Other tokens IDS
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
